### ISSUE TITLE
[BUG] Contract after simple expression function fails

### ISSUE THREAD

**Issue by JohelEGP** [hsutter/cppfront#356]
It works with an extra `;`, if the expression function has a _compound-statement_, or with a statement in-between.

**Minimal reproducer** (https://godbolt.org/z/TfajzxMT1):
```Cpp2
main: () = {
  :() = 1;
  [[assert: 1]]
}
```

**Commands**:
```bash
cppfront x.cpp2
```

**Expected result**: The same as any of the first two workarounds above.

**Actual result and error**:

```output
main.cpp2(3,4): error: subscript expression [ ] must not be empty (if you were trying to name a C-style array type, use 'std::array' instead) (at '[')
```

**Comment by filipsajdak** [comment]
I have a fix for that. No impact on regression tests. Last tests are running and will push it.

**Comment by filipsajdak** [comment]
@JohelEGP fix: #359
